File: 6_protected_attribute.cpp
#include<bits/stdc++.h>

using namespace std;
#define fast ios_base::sync_with_stdio(0); cin.tie(NULL);
#define ll long long

class A{

protected:
    int age;

};

class B:public A{
public:
    B(int age)
    {
        this->age = age;
    }


    void display()
    {
        cout<<"Age -> "<<age<<endl;
    }

};

void solve()
{
    // A obj;

    // obj.age = 19;
    // cout<<obj.age<<endl;

    B newObj(30);
    newObj.display();


}

int main()
{
    fast;
    solve();
    return 0;
}

File: 6_protected_attribute.py
# In python protected not make sense like cpp

class A:
    _age = 0
    
    def display(self):
        print(self._age)
    
    
class B(A):
    def __init__(self,age):
        self._age = age
        
    def display(self):
        print(self._age)
    

obj = A()

obj._age = 60

obj.display()